static/app/components/organizations/projectPageFilter/index.tsx

```diff
@@ -336,17 +336,19 @@ export function ProjectPageFilter({
     )}em`;
   }, [options, desynced]);
 
-  const selectionLimitExceeded = useMemo(
-    () => value.length > SELECTION_COUNT_LIMIT,
-    [value]
-  );
+  const [stagedValue, setStagedValue] = useState<number[]>(value);
+  const selectionLimitExceeded = useMemo(() => {
+    const mappedValue = mapNormalValueToURLValue(stagedValue);
+    return mappedValue.length > SELECTION_COUNT_LIMIT;
+  }, [stagedValue, mapNormalValueToURLValue]);
 
   const menuFooterMessage = useMemo(() => {
     if (selectionLimitExceeded) {
       return hasStagedChanges =>
         hasStagedChanges
-          ? t(
-              'Only up to 50 projects can be selected at a time. You can still press “Clear” to see all projects.'
+          ? tct(
+              'Only up to [limit] projects can be selected at a time. You can still press “Clear” to see all projects.',
+              {limit: SELECTION_COUNT_LIMIT}
```

Comment:
we had some feedback to remove "only"
```suggestion
              'Up to [limit] projects can be selected at a time. You can still press “Clear” to see all projects.',
              {limit: SELECTION_COUNT_LIMIT}
```